### PR TITLE
Vsftpd multimachine test

### PR DESCRIPTION
**Pull request by dzedro**
- Related ticket: https://progress.opensuse.org/issues/43106
- Verification run:
[15SP1](http://10.100.12.155/tests/8836)
[12SP3](http://10.100.12.155/tests/8838)

### CODE REVIEW

File: tests/network/vsftpd.pm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    # copy ssh keys
+    assert_script_run 'ssh-keygen -f /root/.ssh/id_rsa -N ""';
+    type_string "ssh-copy-id -o StrictHostKeyChecking=no \$SERVER\n";
+    wait_still_screen 3;
+    type_password;
+    wait_still_screen 3;
+    send_key 'ret';
+    wait_still_screen 3;
+    type_string "ssh-copy-id -o StrictHostKeyChecking=no \$CLIENT\n";
+    wait_still_screen 3;
```
Comment (kravciak) [inline tests/network/vsftpd.pm]:
```suggestion
    assert_screen 'ssh-password-prompt';
```

File: tests/network/vsftpd.pm [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    type_password;
+    wait_still_screen 3;
```
Comment (kravciak) [inline tests/network/vsftpd.pm]:
```suggestion
```

**Comment by dzedro** [comment]
@kravciak it workds, but travis doesn't like it

File: tests/network/vsftpd.pm [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    # export slenkins variables
+    assert_script_run 'export SERVER=10.0.2.101';
+    assert_script_run 'export CLIENT=10.0.2.102';
+    zypper_call 'in vsftpd';
+    # copy ssh keys
+    assert_script_run 'ssh-keygen -f /root/.ssh/id_rsa -N ""';
+    type_string "ssh-copy-id -o StrictHostKeyChecking=no \$SERVER\n";
+    wait_still_screen 3;
+    type_password;
+    wait_still_screen 3;
```
Comment (kravciak) [inline tests/network/vsftpd.pm]: Fingers crossed and remove
```suggestion
```

File: tests/network/vsftpd.pm [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+use base 'consoletest';
+use testapi;
+use strict;
+use lockapi;
+use utils 'zypper_call';
+
+sub run {
+    select_console 'root-console';
+    if (check_var('HOSTNAME', 'server')) {
```
Comment (kravciak) [inline tests/network/vsftpd.pm]: I think this barrier can be replaced with simple `wait_for_children` call at the end

File: tests/network/vsftpd.pm [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    type_password;
+    send_key 'ret';
+    wait_still_screen 3;
+    assert_script_run 'cd /tmp';
+    assert_script_run 'wget ' . data_url('qam/vsftpd.tar.gz');
+    assert_script_run 'tar xzfv vsftpd.tar.gz';
+    if (check_var('HOSTNAME', 'server')) {
+        assert_script_run 'bash run.sh', 300;
+    }
+    barrier_wait('VSFTPD_FINISHED');
```
Comment (kravciak) [inline tests/network/vsftpd.pm]: I think this barrier can be replaced with simple wait_for_children call at the end
use mmapi 'wait_for_children';
```suggestion
    wait_for_children if check_var('HOSTNAME', 'server');
```

Comment (dzedro) [reply tests/network/vsftpd.pm]: I will leave barrier there, for `wait_for_children` I have to first use get_children at the end it's not simpler than barrier

Comment (kravciak) [reply tests/network/vsftpd.pm]: You don't need get_children first (or I missed something) - anyway, keep whatever you prefer :)

Comment (dzedro) [reply tests/network/vsftpd.pm]: problem is that children end before server is done with testing, barrier is fine

Comment (kravciak) [reply tests/network/vsftpd.pm]: Seems I misunderstood, sorry - I thought client is the one doing testing and server should wait for it..

**Pull request merged**